nav2_behavior_tree/test/plugins/decorator/test_speed_controller.cpp

```diff
@@ -0,0 +1,163 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <chrono>
+#include <memory>
+#include <set>
+
+#include "rclcpp/rclcpp.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "nav2_util/robot_utils.hpp"
+
+#include "../../test_transform_handler.hpp"
+#include "../../test_dummy_tree_node.hpp"
+#include "../../../plugins/decorator/speed_controller.hpp"
+
+using namespace std::chrono;  // NOLINT
+using namespace std::chrono_literals;  // NOLINT
+
+class SpeedControllerTestFixture : public ::testing::Test
+{
+public:
+  static void SetUpTestCase()
+  {
+    transform_handler_ = new nav2_behavior_tree::TransformHandler();
+    config_ = new BT::NodeConfiguration();
+    dummy_node_ = new nav2_behavior_tree::DummyNode();
+
+    // Create the blackboard that will be shared by all of the nodes in the tree
+    config_->blackboard = BT::Blackboard::create();
+    // Put items on the blackboard
+    config_->blackboard->set<rclcpp::Node::SharedPtr>(
+      "node",
+      rclcpp::Node::SharedPtr(transform_handler_));
+    config_->blackboard->set<std::shared_ptr<tf2_ros::Buffer>>(
+      "tf_buffer",
+      transform_handler_->getBuffer());
+    config_->blackboard->set<std::chrono::milliseconds>(
+      "server_timeout",
+      std::chrono::milliseconds(10));
+    config_->blackboard->set<bool>("path_updated", false);
+    config_->blackboard->set<bool>("initial_pose_received", false);
+
+    transform_handler_->activate();
+    transform_handler_->waitForTransform();
+  }
+
+  static void TearDownTestCase()
+  {
+    transform_handler_->deactivate();
+    delete transform_handler_;
+    delete config_;
+    delete dummy_node_;
+    transform_handler_ = nullptr;
+    config_ = nullptr;
+    dummy_node_ = nullptr;
+  }
+
+  void SetUp()
+  {
+    node_ = new nav2_behavior_tree::SpeedController("speed_controller", *config_);
+    node_->setChild(dummy_node_);
+  }
+
+  void TearDown()
+  {
+    node_ = nullptr;
+  }
+
+protected:
+  static nav2_behavior_tree::TransformHandler * transform_handler_;
+  static BT::NodeConfiguration * config_;
+  static nav2_behavior_tree::SpeedController * node_;
+  static nav2_behavior_tree::DummyNode * dummy_node_;
+};
+
+nav2_behavior_tree::TransformHandler * SpeedControllerTestFixture::transform_handler_ = nullptr;
+BT::NodeConfiguration * SpeedControllerTestFixture::config_ = nullptr;
+nav2_behavior_tree::SpeedController * SpeedControllerTestFixture::node_ = nullptr;
+nav2_behavior_tree::DummyNode * SpeedControllerTestFixture::dummy_node_ = nullptr;
+
+TEST_F(SpeedControllerTestFixture, test_behavior)
```

Comment:
A comment here about what this is doing would be good. Because I was reviewing the PR I'm familiar with it enough to pick up on the fact that those sleep durations are specifically meaningful relative the to the odometry speed, but that might not be immediately clear.